### PR TITLE
Integration test CI

### PR DESCRIPTION
**Pull request by wangpatrick57**
**Summary**: Added a CI for running integration tests. Fixed a few bugs to get it working.

**Demo**:
[Successful CI](https://github.com/cmu-db/dbgym/actions/runs/10242135761/job/28331697565)
Screenshot 2024-08-04 at 21 08 29

**Details**:
* Made the runner use the `runner` user instead of `root` so we can start Postgres in it.
* Made new DBGym config for integration test so that the ports and workspace don't conflict (yay config file!). This is not currently necessary since we run it in a Docker container but it's a good precaution.
* In the integration test, we run the full chain (with replay) twice so that we test it both without and with cached results in the workspace. We make sure to delete the workspace before running twice (again not necessary because we're using Docker but is a good precaution).

### CODE REVIEW

File: creations.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+created
```
Comment (lmwnshn) [inline creations.txt:1]: What's this for?

File: hi
Comment (lmwnshn) [inline hi:1]: hi

**Review by lmwnshn — APPROVED** [review]
Some small leftovers. Nice work!

File: hi
Comment (wangpatrick57) [reply hi:1]: hi

File: creations.txt (hunk 1 above)
Comment (wangpatrick57) [reply creations.txt:1]: I had a commit deleting that but I forgot to push it lol

**Pull request merged**